widgets/secretProviders/src/UpdateSecretProviderButton.tsx

```diff
@@ -0,0 +1,116 @@
+import { isEmpty } from 'lodash';
+import type { Manager } from 'cloudify-ui-components/toolbox';
+import { translateSecretProviders, translateForm, validateModalForm } from './widget.utils';
+import type { SecretProvidersWidget } from './widget.types';
+
+const { Icon, Modal, Button, Form } = Stage.Basic;
+
+const { useBoolean, useErrors, useInput } = Stage.Hooks;
+
+interface UpdateSecretProviderButtonProps {
+    secretProvider: SecretProvidersWidget.DataItem;
+    manager: Manager;
+    onSubmit: () => void;
+}
+
+const translateUpdateModal = Stage.Utils.composeT(translateSecretProviders, 'updateModal');
+
+const UpdateSecretProviderButton = ({ secretProvider, manager, onSubmit }: UpdateSecretProviderButtonProps) => {
+    const { errors, setErrors, clearErrors } = useErrors();
+
+    const [hostname, setHostname] = useInput(secretProvider?.connection_parameters?.host ?? '');
+    const [authorizationToken, setAuthorizationToken] = useInput(secretProvider?.connection_parameters?.token ?? '');
+    const [defaultPath, setDefaultPath] = useInput(secretProvider?.connection_parameters?.path ?? '');
```

Comment:
WDYT about the following deduplication? 

```suggestion
   const connectionParameters = secretProvider?.connection_parameters;

    const [hostname, setHostname] = useInput(connectionParameters?.host ?? '');
    const [authorizationToken, setAuthorizationToken] = useInput(connectionParameters?.token ?? '');
    const [defaultPath, setDefaultPath] = useInput(connectionParameters?.path ?? '');
```

Reply:
good idea, implemented: https://github.com/cloudify-cosmo/cloudify-stage/pull/2351/commits/4e6cc0273c14b7a801a6a7313111c1eb589ae68c